src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -1043,13 +1043,14 @@ private boolean shouldDumpItem(
             returnStack = InventoryUtils.addItemStackToProviderWithResult(buildingWorker.getTileEntity(), tempStack);
         }
 
-        new InvWrapper(worker.getInventoryCitizen()).extractItem(slot, stack.stackSize - amountToKeep, false);
-
         if (returnStack == null)
         {
+            new InvWrapper(worker.getInventoryCitizen()).extractItem(slot, stack.stackSize - amountToKeep, false);
```

Comment:
I can do that if you really want to. As stated i was planning a second PR when i have a bit more time that complements this one and turns all our own inventories into IItemHandlers instead of IInventories. Which would make that method depreciated.